packages/react/src/components/DataTable/DataTable.js

```diff
@@ -136,8 +142,84 @@ export default class DataTable extends React.Component {
     this.state = {
       ...getDerivedStateFromProps(props, {}),
       isExpandedAll: false, // Start with collapsed state, treat `undefined` as neutral state
+      colWidth: {},
     };
+    // create refs for the columns and set initial column width
+    this.colRefs = props.headers
+      .map(e => e.key)
+      .reduce((r, e) => {
+        r[e] = createRef();
+        return r;
+      }, {});
     this.instanceId = getInstanceId();
+    this.updateColumnWidthState = this.updateColumnWidthState.bind(this);
```

Comment:
I believe we can use the experimental class fields syntax to avoid these `bind` statements

```suggestion

```

Reply:
done